tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/SeqModelChecker.scala

```diff
@@ -52,12 +54,23 @@ class SeqModelChecker[ExecutorContextT](
     checkerInput.rootModule.assumeDeclarations.foreach { d =>
       trex.assertState(d.body)
     }
-    // apply the Init predicate
-    makeStep(isNext = false, checkerInput.initTransitions)
-    // unroll the transition relation
-    while (searchState.canContinue && trex.stepNo <= params.stepsBound) {
-      // apply the Next predicate
-      makeStep(isNext = true, checkerInput.nextTransitions)
+    val constSnapshot = trex.snapshot()
+
+    while (searchState.canContinue) {
+      // apply the Init predicate
+      makeStep(isNext = false, checkerInput.initTransitions)
+      // unroll the transition relation
+      while (searchState.canContinue && trex.stepNo <= params.stepsBound) {
+        // apply the Next predicate
+        makeStep(isNext = true, checkerInput.nextTransitions)
+      }
+
+      searchState.onRunDone()
+      if (searchState.canContinue) {
+        trex.recover(constSnapshot)
+        logger.info("----------------------------")
+        logger.info(s"Symbolic runs left: ${searchState.nRunsLeft}")
+      }
     }
```

Comment:
I had to read this bit very carefully. Please add a comment that `searchState.canContinue` is now defined differently, and terminates the loop in 1 step for non-simulation, because it's not clear just from this code diff.

Reply:
I agree. We should add a similar comment to the nested `if (searchState.canContinue)`.

Reply:
Added a few comments